src/bosh_azure_cpi/lib/cloud/azure/disk/config_disk_manager.rb

```diff
@@ -0,0 +1,123 @@
+        config_disk_file_path = config_disk_file.path
+
+        mk_image_file_cmd = "dd if=/dev/zero of=#{config_disk_file_path} bs=#{MEGA_SIZE} count=#{CONFIG_DISK_SIZE}"
+        command_runner.run_command(mk_image_file_cmd)
+
+        mkfs_cmd = "mkfs.ext4 -F #{config_disk_file_path} -L #{BOSH_CONFIG_DISK_LABEL}"
+        command_runner.run_command(mkfs_cmd)
+
+        mounted_dir = Dir.mktmpdir(CONFIG_DISK_MOUNT_POINT)
+
+        mount_cmd = "sudo mount -o loop #{config_disk_file_path} #{mounted_dir}"
+        command_runner.run_command(mount_cmd)
+
+        user = ENV['USER']
+        chmod_cmd = "sudo chown #{user} #{mounted_dir}"
+        command_runner.run_command(chmod_cmd)
+
+        mkdir_cmd = "mkdir -p #{mounted_dir}/#{CONFIGS_RELATIVE_PATH}"
+        command_runner.run_command(mkdir_cmd)
+
+        meta_data_str = JSON.dump(meta_data_obj)
+        File.open("#{mounted_dir}/#{CONFIGS_RELATIVE_PATH}/MetaData", 'w') do |file|
+          file.write(meta_data_str)
+        end
+
+        user_data_str = JSON.dump(user_data_obj)
+        File.open("#{mounted_dir}/#{CONFIGS_RELATIVE_PATH}/UserData", 'w') do |file|
+          file.write(user_data_str)
+        end
+
+        unmount_cmd = "sudo umount #{mounted_dir}"
+        command_runner.run_command(unmount_cmd)
+        umounted = true
+        # convert the plain file to vhd
+        # TODO check whether qemu-utils is installed or not. if not, install it.
+        config_disk_vhd_file = Tempfile.new(CONFIG_DISK_FILE_PATH_PREFIX)
+        config_disk_vhd_file.close
+        config_disk_vhd_file_path = config_disk_vhd_file.path
+
+        convert_to_vhd_cmd = "qemu-img convert -O vpc -o subformat=fixed #{config_disk_file_path} #{config_disk_vhd_file_path}"
+        command_runner.run_command(convert_to_vhd_cmd)
+
+        disk_name = "#{MANAGED_CONFIG_DISK_PREFIX}-#{vm_name}.vhd"
+        @blob_manager.create_page_blob(
+          @storage_account_manager.default_storage_account_name,
+          CONFIG_DISK_CONTAINER,
+          config_disk_vhd_file_path,
+          disk_name,
+          {}
+        )
+        page_blob_created = true
+        config_disk_file_uri = @blob_manager.get_blob_uri(@storage_account_manager.default_storage_account_name, CONFIG_DISK_CONTAINER, disk_name)
+        disk_id = DiskId.create('None', true, disk_name: disk_name, resource_group_name: resource_group_name)
+
+        @disk_manager2.create_disk_from_blob(disk_id, config_disk_file_uri, location, STORAGE_ACCOUNT_TYPE_STANDARD_LRS)
+
+        disk = @disk_manager2.get_data_disk(disk_id)
+        @logger.info("disk created: #{disk}")
+        # TODO: defer one task to clean up the config disk.
+        #       we need to acquire the lock for the resources we are operating too.
+        disk
+      rescue StandardError => e
+        @logger.error("Failed to prepare the config disk, Error: #{e.inspect}\n#{e.backtrace.join("\n")}")
+        cloud_error('Failed to prepare the config disk')
```

Comment:
cloud_error already contains logger.error.  you can delete line 104.

Reply:
but no detail error message.

Reply:
cloud_error("Failed to prepare the config disk, Error: #{e.inspect}\n#{e.backtrace.join("\n")}") will be ok. otherwise you will see `Failed to prepare the config disk` twice.